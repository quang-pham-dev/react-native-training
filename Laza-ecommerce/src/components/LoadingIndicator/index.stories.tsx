import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from 'storybook/stories/CenterView';
import LoadingIndicator from './index';

storiesOf('LoadingIndicator', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('large size', () => <LoadingIndicator size='large' />)
  .add('small size', () => <LoadingIndicator size='small' />);
