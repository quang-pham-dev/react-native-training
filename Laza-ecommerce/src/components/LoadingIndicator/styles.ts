import { StyleSheet } from 'react-native';

// Themes
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  loading: {
    margin: Metrics.margin.md,
  },
});

export default styles;
