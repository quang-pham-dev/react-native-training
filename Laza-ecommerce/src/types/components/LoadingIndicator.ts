export interface ILoadingIndicatorProps {
  size?: number | 'small' | 'large' | undefined;
  color?: string;
}
