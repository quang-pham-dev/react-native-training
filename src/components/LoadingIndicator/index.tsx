import React from 'react'
import {ActivityIndicator} from 'react-native'

// Styles
import ViewStyled from '@components/View/View.styles'

// Themes
import {Colors} from '@themes'

export enum LoadingSize {
  SMALL = 'small',
  LARGE = 'large',
}

type LoadingIndicatorProps = {
  size?: LoadingSize | undefined
  color?: string
}

const LoadingIndicator = ({
  size,
  color = Colors.palette.white,
}: LoadingIndicatorProps) => (
  <ViewStyled.Custom>
    <ActivityIndicator size={size} color={color} />
  </ViewStyled.Custom>
)

export default LoadingIndicator
