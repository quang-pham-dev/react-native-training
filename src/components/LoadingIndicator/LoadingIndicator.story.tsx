import * as React from 'react'
import {storiesOf} from '@storybook/react-native'
import {StoryScreen, Story, UseCase} from '../../../storybook/views'
import LoadingIndicator, {LoadingSize} from './index'
import {select} from '@storybook/addon-knobs'
import {Colors} from '@themes'

declare let module: NodeModule

const typesOption = [LoadingSize.SMALL, LoadingSize.LARGE]
const color = {...Colors}

storiesOf('LoadingIndicator', module)
  .addDecorator(fn => <StoryScreen>{fn()}</StoryScreen>)
  .add('Small', () => {
    const colorPick = select('color', Object.keys(color), 'whiteNavigation')
    return (
      <Story>
        <UseCase bgColor="#2a9fd8" text="Default">
          <LoadingIndicator
            size={select('size', typesOption, typesOption[0])}
            color={colorPick}
          />
        </UseCase>
      </Story>
    )
  })
  .add('Large', () => {
    const colorPick = select('color', Object.keys(color), 'red')
    return (
      <Story>
        <UseCase bgColor="#2a9fd8" text="Large">
          <LoadingIndicator
            size={select('size', typesOption, typesOption[0])}
            color={colorPick}
          />
        </UseCase>
      </Story>
    )
  })
