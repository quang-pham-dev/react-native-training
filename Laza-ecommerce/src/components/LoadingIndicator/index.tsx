import React, { memo } from 'react';
import { ActivityIndicator, SafeAreaView } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Types
import { ILoadingIndicatorProps } from 'types/components/LoadingIndicator';
import { LOADING_SIZE } from 'types/common/Enums';

// Themes
import Colors from 'themes/Colors';

// Styles
import styles from './styles';

const LoadingIndicator = ({
  size = LOADING_SIZE.LARGE,
  color = Colors.secondaryColor,
}: ILoadingIndicatorProps) => (
  <SafeAreaView>
    <ActivityIndicator style={styles.loading} size={size} color={color} />
  </SafeAreaView>
);

export default memo(LoadingIndicator, isEqual);
