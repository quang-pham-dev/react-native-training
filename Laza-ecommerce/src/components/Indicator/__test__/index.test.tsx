import React from 'react';
import LoadingIndicator from 'components/Indicator';
import renderer from 'react-test-renderer';

describe('LoadingIndicator Component', () => {
  test('should render correctly', () => {
    const tree = renderer.create(<LoadingIndicator loadingSize='large' />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render color LoadingIndicator correctly', () => {
    const tree = renderer.create(<LoadingIndicator loadingSize='small' color='red' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
