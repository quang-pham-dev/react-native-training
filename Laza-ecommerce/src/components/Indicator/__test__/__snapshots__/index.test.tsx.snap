// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator Component should render color LoadingIndicator correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="red"
  hidesWhenStopped={true}
  size="small"
  style={
    Object {
      "margin": 13,
    }
  }
/>
`;

exports[`LoadingIndicator Component should render correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#FFC107"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "margin": 13,
    }
  }
/>
`;
